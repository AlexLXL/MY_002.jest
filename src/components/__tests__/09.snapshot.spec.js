import snapshot from "../09snapshotTest";
import snapshotDynamic from "../09snapshotTestDynamic";
import { shallowMount } from "@vue/test-utils";

describe("09snapshotTest", () => {
    test("No.001 测试创建快照(静态页面)", () => {
        const wrapper = shallowMount(snapshot);
        expect(wrapper.element).toMatchSnapshot();
        // npm run test:unit, 第1次是创建快照, 第2次开始是对比快照
        // npm run test:unit -- --updateSnapshot, 更新所有快照
        // npm run test:unit -- --u, 更新所有快照
        // npm run test:unit -- --w, 交互方式更新快照, 按<u>更新失败的快照
    })

    test("No.002 测试创建快照(动态页面)(props使用类型A)", () => {
        // 将动态页面(props、data、date.now等)转换为可确定的静态页面
        const wrapper = shallowMount(snapshotDynamic, {
            propsData: {
                type: "A"
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    })

    test("No.003 测试创建快照(动态页面)(props使用类型B)", () => {
        // 将动态页面(props、data、date.now等)转换为可确定的静态页面
        const wrapper = shallowMount(snapshotDynamic, {
            propsData: {
                type: "B"
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    })
})