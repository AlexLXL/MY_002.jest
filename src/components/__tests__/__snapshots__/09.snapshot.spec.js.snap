// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`09snapshotTest No.001 测试创建快照(静态页面) 1`] = `
<div>
  <header>
    09.快照测试 [静态组件]
  </header>
   
  <p>
    获取代码的快照，并将其与以前保存的快照进行对比 (jest是通过对比序列化值)
  </p>
   
  <transition-stub>
    <svg
      class="spinner"
      height="50px"
      viewBox="0 0 50 50"
      width="50px"
    >
      <circle
        class="path"
        cx="22"
        cy="22"
        fill="none"
        r="20"
        stroke-linecap="round"
        stroke-width="4"
      />
    </svg>
  </transition-stub>
</div>
`;

exports[`09snapshotTest No.002 测试创建快照(动态页面)(props使用类型A) 1`] = `
<div>
  <header>
    09.快照测试 [动态组件]
  </header>
   
  <div>
    AAAAA组件
  </div>
   
  <!---->
</div>
`;

exports[`09snapshotTest No.003 测试创建快照(动态页面)(props使用类型B) 1`] = `
<div>
  <header>
    09.快照测试 [动态组件]
  </header>
   
  <!---->
   
  <div>
    BBBBB组件
  </div>
</div>
`;
